src/node.rs

```diff
@@ -78,6 +78,8 @@ impl PbftNode {
         msg: ParsedMessage,
         state: &mut PbftState,
     ) -> Result<(), PbftError> {
+        info!("{}: Got peer message: {}", state, msg.info());
```

Comment:
This commit should be reworded to explain what defects are being fixed. What is interesting to the reader is not that the test now passes, but what was broken before that you have fixed. This might need to be broken up into several commits to accomplish that, if there is more than one fix contained here.

Reply:
@aludvik: Does it suffice to say that these fixes are really just implementing node catchup part II due to rushing the previous code out the door for a deadline? They would've been squashed into the previous PR if there had been time.

Reply:
I think it is important that the commit message be helpful in understanding what the commit is doing and why. "Fix CFT tests" gives me the impression the commit is modifying the tests. In reality it is making changes to the implementation in order to pass the tests, fixing a number of bugs in the process, so I think it is misleading.

To give you something more concrete, based on the commit description, I would suggest having one commit like this:
```
Synchronize sequence number on new block

Ensures that the sequence number doesn't get out of sync with actual state by
syncing them whenever a new block is received. This is required to pass the
liveness test.
```

and another like this:
```
Handle missing state after crash in catch up

Provides a more hardy version of catchup that is tolerant of booting up 
without any old state after a crash. This is required to pass the liveness test.
```